src/App.js

```diff
@@ -1,82 +1,186 @@
 import React from 'react';
 import './App.css'
+import axios from 'axios'
 
 class App extends React.Component {
+  constructor () {
+    super();
+    this.state = {
+      name: '',
+      email: '',
+      password: '',
+      file: '',
+      number: '',
+      age: '',
+      gender: '',
+      hobbies: {duathuyen: false, dabong: false, caulong: false},
+      note: ''
+    }
+  }
+
+  handldeChange = (event) => {
+    const {name,type} = event.target
+    let value = type === 'checkbox' ? event.target.checked : event.target.value;
+    // if (type === 'checkbox') {
+    //   const checkbox = this.state[name]
+    //   checkbox[value] = !checkbox[value]
+    //   this.setState({
+    //     [name]: checkbox,
+    //   })
+    //   return
+    //   }
+
+    this.setState ({
+      [name]: value
+    });
+  }
+
+  handleSubmit = (event) => {
```

Comment:
Nếu chỉ có 1 param thì có thể bỏ dấu `(` và `)` đi:
```suggestion
  handleSubmit = event => {
```